homeassistant/components/google/calendar.py

```diff
@@ -37,23 +39,51 @@
     DOMAIN,
     SERVICE_SCAN_CALENDARS,
 )
-from .const import DISCOVER_CALENDAR
+from .const import (
+    CONF_CALENDAR_ACCESS,
+    DISCOVER_CALENDAR,
+    EVENT_DESCRIPTION,
+    EVENT_END_CONF,
+    EVENT_END_DATE,
+    EVENT_END_DATETIME,
+    EVENT_IN,
+    EVENT_IN_DAYS,
+    EVENT_IN_WEEKS,
+    EVENT_START_CONF,
+    EVENT_START_DATE,
+    EVENT_START_DATETIME,
+    EVENT_SUMMARY,
+    EVENT_TYPES_CONF,
+    FeatureAccess,
+)
 
 _LOGGER = logging.getLogger(__name__)
 
-DEFAULT_GOOGLE_SEARCH_PARAMS = {
-    "orderBy": "startTime",
-    "singleEvents": True,
-}
-
 MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=15)
 
 # Events have a transparency that determine whether or not they block time on calendar.
 # When an event is opaque, it means "Show me as busy" which is the default.  Events that
 # are not opaque are ignored by default.
-TRANSPARENCY = "transparency"
 OPAQUE = "opaque"
 
+_EVENT_IN_TYPES = vol.Schema(
+    {
+        vol.Exclusive(EVENT_IN_DAYS, EVENT_TYPES_CONF): cv.positive_int,
+        vol.Exclusive(EVENT_IN_WEEKS, EVENT_TYPES_CONF): cv.positive_int,
+    }
+)
+
+SERVICE_CREATE_EVENT = "create_event"
+CREATE_EVENT_SCHEMA = {
+    vol.Required(EVENT_SUMMARY): cv.string,
+    vol.Optional(EVENT_DESCRIPTION, default=""): cv.string,
+    vol.Exclusive(EVENT_START_DATE, EVENT_START_CONF): cv.date,
+    vol.Exclusive(EVENT_END_DATE, EVENT_END_CONF): cv.date,
+    vol.Exclusive(EVENT_START_DATETIME, EVENT_START_CONF): cv.datetime,
+    vol.Exclusive(EVENT_END_DATETIME, EVENT_END_CONF): cv.datetime,
+    vol.Exclusive(EVENT_IN, EVENT_START_CONF, EVENT_END_CONF): _EVENT_IN_TYPES,
```

Comment:
Only a single exclusion group is allowed as far as I know.

https://github.com/alecthomas/voluptuous/blob/2466ee2ffe901cd3df3f8675c1f21b4786b5b58e/voluptuous/schema_builder.py#L1044-L1080

Reply:
Ok, this is just copy paste the existing schema (I don't even know what exclusion even means! :) )  I'll have a closer look.

Reply:
I see that more than one group seems to behave as expected:
```
>>> schema = vol.Schema({
    vol.Exclusive('alpha', 'angles'): int,
    vol.Exclusive('beta', 'angles'): int,
    vol.Exclusive('square', 'shapes'): int,
    vol.Exclusive('circle', 'shapes'): int
})
>>> schema({'alpha': 30, 'circle': 45})
{'alpha': 30, 'circle': 45}
>>> schema({'square': 30, 'circle': 45})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/allen/home-assistant-core/venv/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 272, in __call__
    return self._compiled([], data)
  File "/home/allen/home-assistant-core/venv/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 577, in validate_dict
    raise er.MultipleInvalid(errors)
voluptuous.error.MultipleInvalid: two or more values in the same group of exclusion 'shapes' @ data[<shapes>]
```

Am I understanding your comment correctly or are you saying something dfiferent?

Reply:
Yes, this line has three parameters passed to Exclusive instead of two. The third parameter will be a custom message used when validation fails.